realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -293,6 +297,29 @@ public int hashCode() {
         }
     }
 
+    /**
+     * Waits for change committed by {@link SharedGroup} in other Thread.
+     *
+     * @return {@code true} if successfully detects change, {@code false} no change has detected otherwise.
+     */
+    public boolean waitForChange() {
+        if (isStoppedWaitingByUser) {
+            isStoppedWaitingByUser = false;
+            return false;
+        }
+        boolean changeStatus = nativeWaitForChange(nativePtr);
+        isStoppedWaitingByUser = false;
+        return changeStatus;
+    }
+
+    /**
+     * Stops waiting for change.
+     */
+    public void stopWaitForChange() {
+        isStoppedWaitingByUser = true;
```

Comment:
I think this should be set after the call to `nativeStopWaitForChange(nativePtr);` in case the Core call crash we shouldn't modify the flag state